src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7203,8 +7472,23 @@ public override BoundNode VisitFieldAccess(BoundFieldAccess node)
 
         public override BoundNode VisitPropertyAccess(BoundPropertyAccess node)
         {
-            var updatedMember = VisitMemberAccess(node, node.ReceiverOpt, node.PropertySymbol);
-            SetUpdatedSymbol(node, node.PropertySymbol, updatedMember);
+            var property = node.PropertySymbol;
+            var updatedMember = VisitMemberAccess(node, node.ReceiverOpt, property);
+
+            if (!IsAnalyzingAttribute &&
+                (property.IsStatic || node.ReceiverOpt is BoundThisReference || node.ReceiverOpt is BoundBaseReference))
+            {
```

Comment:
Where is the code that implements this (changing the state for a member of some object when a method is called on that object) for when the receiver is something other than `this`?